schema/deploy/warehouse/identifier/identifier-type.sql

```diff
@@ -0,0 +1,12 @@
+-- Deploy seattleflu/schema:warehouse/identifier/identifier-type to pg
+-- requires: warehouse/identifier
+
+begin;
+
+alter table warehouse.identifier_set
+    add column identifier_type text;
```

Comment:
Do we want to make the type of this column more restricted than `text`? I think we might. Much of the ID3C core code, not just our customizations, makes a distinction between sample and collection, for instance, so it feels like that should be formalized conceptually by the schema.

If we do want to narrow down the type, Pg's domain types are a good option for this sort of value where a vocabulary table is not necessary (i.e. because it's not intended to be customized). There's an example of using domain types for `warehouse.individual.sex`.

(If it was intended and expected to be customized or extensible, then a new vocabulary table would be more appropriate. An example of this in ID3C is the `warehouse.encounter_location_relation` table.)

---

Similarly, should we make this column `not null`? Probably, IMO. I think we can exhaustively assign our existing identifier sets. This will require a bit of coordination by making it `not null` in a separate sqitch change, so we can interleave with a manual `update`.

---

The `id3c identifier set create` command should also be able to set this column so we can do so when making new sets later. This is particularly important if the column is not null, but also to avoid needing manual `insert` or `update` statements in addition to the CLI command.

---

Let's index this column, as it seems very likely we'll be doing filtering/searching based on it. (I was reminded to consider this by the [schema development guidelines](https://github.com/seattleflu/id3c/blob/master/README.md#guidelines) I wrote long ago. :-)

--- 

Separately, the repetition of the table name in the column names is a pattern avoided in ID3C's schema (with the sole exception of generated serial pks). Let's name the column just `type` or `kind` or `use`? (I like `use` because it's more specific and less overloaded than `type`, but this is mostly aesthetics.)

Reply:
Good suggestions, I'll get started on these! And thanks for the link to the guidelines, those are helpful.

I thought a new 1:M table might be overkill, but if the list of identifier types is likely to grow then that seems like a good option. It would also allow us to use an integer value instead of text, which should perform better for search/filter and even more so with indexing.

I wasn't sure how to implement `NOT NULL` without the required values in place, but a separate sqitch change makes sense.

The changes needed on `id3c identifier set create` is what I was getting at on the call earlier... whether that should be included in this PR or a later one. But with `NOT NULL` in place it should definitely be included here to avoid breaking it. That will also be the case for the new api function you have in PR #230, [put_identifier_set](https://github.com/seattleflu/id3c/pull/230/files#diff-4423d40718dcdaa9d1a971ec4f113f118e302460a71bc7c3118086d770817d49R201) so merging these two branches into master is going to take some coordination.

I avoided using `type` for the reason you mentioned but I wanted to keep it as part of the column name since that's how it was being referred to in other conversations. If `use` seems intuitive enough I can go with that.

Reply:
After some testing I realized an integer id was not the way to go here. There's too much overhead with the extra join needed to query identifiers when the `use` text value is in a separate table.  I ended up using `warehouse.encounter_location_relation` as a template as @tsibley suggested, using `citext` as the PK which eliminates the need for a join.

I also tested adding an index on the new `use` column of `warehouse.identifier_set` but saw no improvement when querying identifiers by `use` type (using `EXPLAIN ANALYZE` method). I was able to reduce cost of the same queries by approximately 50% by adding a b-tree index on `warehouse.identifier.identifier_set_id`.